docs/user/high-availability.md

```diff
@@ -0,0 +1,130 @@
+
+There are two options for updating the flows on an HA instance.
+
+### Disabling HA mode
+
+If you disable HA mode via the setting page, after the instance has been restarted
+the editor will be available and changes can be made. However, this will mean a
+period of downtime whilst the instance is brought in and out of HA mode.
+
+### DevOps Pipeline
+
+Create a [DevOps pipeline](./devops-pipelines.md) from a 'development' Instance to push updates to your HA
+instance.
+
+This allows you to free develop your flows in one instance and when you are happy
+with the results, use the pipeline to push the changes over to the HA instance.
+
+## Building HA-ready flows
+
+Whilst FlowForge can help run multiple copies of your flows, and provide the
+necessary load balancing between those copies, it still requires the flows to
+have been created with HA in mind.
+
+This guide provides some things to consider when building HA-ready flows. We'll
+continue to expand on this as we gather more feedback from our users.
+
+There are two things to consider - how state is managed and how work is distributed
+between the copies.
+
+### Managing state
+
+The most well-suited flows for HA are those that do not depend on local state being
+maintain between individual messages coming into the flow. This is because, in an HA
+instance, the messages are distributed between the copies - none of them get to see
+every message.
+
+When state is required, it needs to be stored somewhere that all instances can
+access it - for example an external database service.
+
+FlowForge provides a [persistent context](./persistent-context.md) option - however
+this includes a local caching layer that means it doesn't fully synchronize between
+the instances in real time. This is something we'll be working on for a future iteration
+of the feature, as it will require some changes in Node-RED to unlock its full potential.
+
+There is also the state that is implicitly maintained within the nodes of a flow.
+For example, the Smooth node can be used to calculate a running average value of
+messages passing through it. The node does that by keeping in memory the recent
+values so it can recalculate the average with each update. In an HA instance,
+the node will only be calculating the average for the messages it sees.
+
+### Routing work
+
+When HA is enabled, all HTTP traffic directed at the instance will be load-balanced
+between the copies automatically.
+
+It gets a little more complicated where flows connect out to external systems to 
+pull in work - for example, when using MQTT.
+
+MQTT includes a feature called [Shared Subscriptions](https://www.hivemq.com/blog/mqtt5-essentials-part7-shared-subscriptions/)
+that allows a broker to distribute messages between a group of subscribers. This
+provides the load balancing needed for an HA instance - but the flow must be configured
+to use the appropriate shared subscription topics.
```

Comment:
```suggestion
to use the appropriate shared subscription topics, as well as to not set a ClientID on the connection to avoid conflicts between the connections.
```